neurom/apps/morph_stats.py

```diff
@@ -241,23 +260,38 @@ def full_config():
     """Returns a config with all features, all modes, all neurite types."""
     modes = ['min', 'max', 'median', 'mean', 'std']
     return {
-        'neurite': {feature: modes for feature in _NEURITE_FEATURES},
-        'morphology': {feature: modes for feature in _MORPHOLOGY_FEATURES},
-        'population': {feature: modes for feature in _POPULATION_FEATURES},
-        'neurite_type': list(_NEURITE_MAP.keys()),
+        "neurite": [[name, {"kwargs": {}, "modes": modes}] for name in _NEURITE_FEATURES],
+        "morphology": [[name, {"kwargs": {}, "modes": modes}] for name in _MORPHOLOGY_FEATURES],
+        "population": [[name, {"kwargs": {}, "modes": modes}] for name in _POPULATION_FEATURES],
+        "neurite_type": list(_NEURITE_MAP.keys()),
     }
 
 
-def sanitize_config(config):
+def _kwargs_modes_layout(category_features):
+    """Converts a dictionary or iterable of features to a list of feature name and option pairs."""
+    def standardize_options(options):
+        """Returns options as a dict with two keys: 'kwargs' and 'modes'."""
+        # convert from short format
+        if isinstance(options, list):
+            return {"kwargs": {}, "modes": options}
+
+        return {"kwargs": options.get("kwargs", {}), "modes": options.get("modes", [])}
+
+    # either a dict or an iterable of feature_name, options pairs
+    pairs = category_features.items() if isinstance(category_features, dict) else category_features
+
+    return [[name, standardize_options(options)] for name, options in pairs]
+
+
+def _sanitize_config(config):
```

Comment:
I am just wondering if we shouldn't perform a validation step on the config, so we can report issues accurately about why the config is invalid instead of just failing later in the code with a KeyError or something like that. It can come in another PR though.

Reply:
Do you mean using a schema? That's a good idea indeed, but better do it on a follow up PR yeah.

Reply:
Yes that's what I meant but let's do this in another PR